### ISSUE TITLE
Fetch File Server- Adding custom data makes `getFetchFileServerCatalog()` fail.

### ISSUE THREAD

**Issue by Karuhanga** [tonyofrancis/Fetch#213]
@tonyofrancis 
I spent the weekend pulling out my hair over this. A fix is needed. I am still trying to find the source of the issue as well.
The error thrown is here ([https://gist.github.com/Karuhanga/42bf9ea5846e0a3598aff8313fd4f13f](https://gist.github.com/Karuhanga/42bf9ea5846e0a3598aff8313fd4f13f))

**Steps to reproduce;**
Using the sample App, adding any data to the custom data MutableMap. Modified method below;

```
        private void addFileResourceToServer(File file, long fileLength) {
            final FileResource fileResource = new FileResource();
            fileResource.setFile(file.getAbsolutePath());
            fileResource.setName(file.getName());
            fileResource.setId(file.getAbsolutePath().hashCode());
            fileResource.setLength(fileLength);
            fileResource.getCustomData().put("jar", "cookie"); //added custom data
            final String fileMd5 = FetchCoreUtils.getFileMd5String(file.getAbsolutePath());
            if (fileMd5 != null) {
                fileResource.setMd5(fileMd5);
            }
            fetchFileServer.addFileResource(fileResource);

            downloadFileResourceUsingFetch();
            fetch.getFetchFileServerCatalog(getCatalogRequest(),
                result -> Timber.d("Catalog:" + result.toString()),
                error -> Timber.d("Catalog Fetch error:" + error.toString()));
        }

**Comment by tonyofrancis** [comment]
@Karuhanga I just got back from vacation. I will look into this as soon as possible.

**Comment by tonyofrancis** [comment]
@Karuhanga found the bug working on a fix

**Comment by tonyofrancis** [comment]
@Karuhanga  Just released Fetch 2.2.0-RC5 with this fix. Please let me know if it works well for you.

**Comment by Karuhanga** [comment]
@tonyofrancis Awesome. I just checked it out and it is working fine with the sample app. I'll confirm with my project.
Thanks for the good work.
